docs/_getting-started/dev-mode.md

```diff
@@ -1,64 +1,339 @@
+-   Sets max peers to 0 (meaning Geth does not search for peers)
+-   Turns off discovery by other nodes (meaning the node is invisible to other nodes)
+-   Sets the gas price to 0 (no cost to send transactions)
+-   Uses the Clique proof-of-authority consensus engine with which allows blocks to be mined as-needed without excessive CPU and memory consumption
 -   Uses on-demand block generation, producing blocks when transactions are waiting to be mined
 
+This configuration enables developers to experiment with Geth's source code or develop new applications without having to sync to a pre-existing public network. Blocks are only mined when there are pending transactions. Developers can break things on this network without affecting other users. This page will demonstrate how to spin up a local Geth testnet and a simple smart contract will be deployed to it using the Remix online integrated development environment (IDE).
+
+## Prerequisites
+
+It is assumed that the user has a working Geth installation (see [installation guide](../_install-and-build/)).
+It would also be helpful to have basic knowledge of Geth and the Geth console. See [Getting Started](../_getting-started/).
+Some basic knowledge of [Solidity](https://docs.soliditylang.org/) and [smart contract deployment](https://ethereum.org/en/developers/tutorials/deploying-your-first-smart-contract/) would be useful.
+
 ## Start Geth in Dev Mode
 
-You can specify a data directory to maintain state between runs using the `--datadir` option, otherwise, databases are ephemeral and in-memory:
+Starting Geth in developer mode is as simple as providing the `--dev` flag. It is also possible to create a realistic block creation frequency by setting `--dev.period 13` instead of creating blocks only when transactions are pending. There are also additional configuration options required to follow this tutorial. 
+
+First, a data directory will be created so that the local blockchain state can be maintained between sessions - without this the local blockchain is ephemeral and only existing in memory for the duration the node is running. In the command below this is achieved by providing the directory name, `dev-chain` as an argument following the `--datadir` flag. 
+
+Next, `http` must be enabled so that the Javascript console can be attached to the Geth node, and some namespaces must be specified so that certain functions can be executed from the Javascript console, specifically `eth`, `web3` and `personal`. 
+
+Finally, Remix will be used to deploy a smart contract to the node which requires information to be exchanged externally to Geth's own domain. To permit this, the `net` namespace must be enabled and the Remix URL must be provided to `--http.corsdomain`. The full command is as follows:
+
+```shell
+
+geth --datadir dev-chain --dev --http --http.api eth,web3,personal,net --http.corsdomain "http://remix.ethereum.org"
```

Comment:
Is there a reason to use `--datadir` here? As in, does the example rely on reusing the same chain? Otherwise I think let's drop that, it's more common to just use an ephemeral datadir in dev mode IMO

Reply:
the only reason is to facilitate an explanation that including or omitting the flag toggles ephemeral or persistent states between sessions. The example does not require reusing the same chain, so I guess it does not really need to be included in the code block.

Reply:
We also did this in the previous update attempt of the dev mode page: https://github.com/ethereum/go-ethereum/pull/24467/files

Reply:
There are some nuances with reusing the dev mode datadir which may complicated to explain.

When `--datadir` is used with `--dev`, and the keystore already contains an account, geth will fail to start because it cannot unlock the account automatically. This can be fixed using the `--password` flag. @wise4rmgod got stuck on this issue when working on #24467. It might be good to document that somewhere, maybe toward the end of the guide.

Reply:
Maybe a good solution would be to forego the whole datadir reuse in the main tutorial and add a small section at the end explaining how to reuse the test blockchain with `--datadir`.

Reply:
ok, thanks @fjl I updated the page as you suggested in the latest commit